java/src/org/openqa/selenium/firefox/FirefoxDriver.java

```diff
@@ -165,7 +165,7 @@ private FirefoxDriver(
     Optional<URI> reportedUri =
         CdpEndpointFinder.getReportedUri("moz:debuggerAddress", capabilities);
 
-    if (reportedUri.isPresent()) {
+    if (reportedUri.isPresent() && !(Boolean) capabilities.getCapability("webSocketUrl")) {
```

Comment:
Wouldn't this throw if the capability is not present? Maybe we check first if it is present and then we check the boolean value.

Reply:
You mean to do a `null` check first?

Reply:
`caps.is("webSocketUrl");`

Reply:
Looks like a better way, indeed

Reply:
Done